packages/core/analytics/core-analytics-server-internal/src/analytics_service.test.ts

```diff
@@ -7,41 +7,37 @@
  */
 
 import { firstValueFrom, Observable } from 'rxjs';
+import { createTestEnv, createTestPackageInfo } from '@kbn/config-mocks';
 import { mockCoreContext } from '@kbn/core-base-server-mocks';
 import { analyticsClientMock } from './analytics_service.test.mocks';
 import { AnalyticsService } from './analytics_service';
 
+const packageInfo = createTestPackageInfo();
+
+const createCoreContext = () => {
+  const env = createTestEnv({ packageInfo });
+  return mockCoreContext.create({ env });
+};
+
 describe('AnalyticsService', () => {
   let analyticsService: AnalyticsService;
   beforeEach(() => {
     jest.clearAllMocks();
-    analyticsService = new AnalyticsService(mockCoreContext.create());
+    analyticsService = new AnalyticsService(createCoreContext());
   });
 
   test('should register the context provider `build info` on creation', async () => {
     expect(analyticsClientMock.registerContextProvider).toHaveBeenCalledTimes(1);
     await expect(
       await firstValueFrom(analyticsClientMock.registerContextProvider.mock.calls[0][0].context$)
-    ).toMatchInlineSnapshot(
-      {
-        branch: expect.any(String),
-        buildNum: 9007199254740991,
-        buildSha: 'XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX',
-        isDev: true,
-        isDistributable: false,
-        version: expect.any(String),
```

Comment:
Operations already patched the test in 2b0754c48672caf42bf661674fb1bdaabf0c2326 to avoid failures on the 8.5 branch, I modified it to properly work on all branches instead.